content/posts/2020-07-15-tweeting-from-pull-requests/index.md

```diff
@@ -0,0 +1,45 @@
+
+The easiest solution would be sharing the credentials, some may think. That would work but that only wouldn’t be enough. Collective ownership is also about making sure decisions are made and agreed upon. We would need at least people reviewing each other’s content.
+
+So thinking a bit more, what I envisioned was a flow that would allow us to:
+
+1. have someone proposing a new tweet
+2. notify others when there’s new content to be reviewed
+3. have others reviewing and approving it
+4. publish tweet when content is approved
+
+![./diagram.png](diagram.png)
+
+## The solution
+
+Because these reviewers are mostly engineers, the closer this whole process is from their natural habitat, the better. We use GitHub and Slack all day so the solution would definitely be somewhere in there.
+
+I have used [Danger](https://danger.systems/js/) and [Peril](https://github.com/danger/peril) quite a lot in the past. It’s a great tool to automate many of the workflows we have, especially when it comes to review of Pull Requests. In fact, Loadsmart has its own set of [settings](https://github.com/loadsmart/peril-settings) that are used across many of our repositories on GitHub. But this time I decided to give GitHub actions a try.
+
+Surprisingly, most of the things I was planning to implement were already done by [Gregor Martynus](https://github.com/gr2m) in their [twitter-together](https://github.com/gr2m/twitter-together) project. From the steps of the diagram above, the Action takes care of 1, 2 and 4. The only step missing here is the Slack notification.
+
+There’s a few Actions to interact with Slack in the marketplace. They are all very similar to each other and it seems to me that the same problem was solved multiple times (thinking out loud, it would be good if GitHub had a way to make Actions "official"). So I decided to implement this step myself. I would have to create a custom action in the end to glue all the work together (passing to Slack the content of the tweet in the PR) so this would be only an extra step. By the way, their [Block Kit API](https://api.slack.com/block-kit) is very flexible and you can build interesting custom notifications:
+
+![./notification.png](notification.png)
+
+The example above presents a notification for a [Pull Request](https://github.com/loadsmart/tweets/pull/9) that after opened (and [identified](https://github.com/loadsmart/tweets/pull/9/checks?check_run_id=613179583) as a valid tweet-type PR), sends a message to our team asking for reviews. Once approved, it gets merged into [master](https://github.com/loadsmart/tweets/runs/613204451?check_suite_focus=true) and then finally [tweeted](https://twitter.com/LoadsmartEng/status/1253412556867203073).
```

Comment:
What do you think of having a print screen with the Github actions flow?
Or maybe, also, a screenshot of the GitHub action you created. Just to go a bit more in deep of the implementation part.

I know people can see the code since it is open, but maybe it would be nice to have an overview of "how to implement this solution" if a person only reads the blog.

Reply:
Unfortunately, GH doesn't render a nice flow as CircleCI does. So the only thing we could add here is either a screenshot of the execution or the code itself. I think both would be better presented as they are (on GitHub) so readers can interact with it (thus the links to both).

I also wanted to make readers access the repo and get curious about the implementation itself. So these links are simply calls to action.

Reply:
Got the idea now. Then it makes sense 👍  Thanks!